ipa-core/src/cli/test_setup.rs

```diff
@@ -116,6 +117,24 @@ fn sharded_keygen(args: &TestSetupArgs) -> Result<(), BoxError> {
     .flatten()
     .collect::<Vec<_>>();
 
+    // for match key encryption keys we need to do some extra work. All shards
+    // must have access to the same set of encryption keys in order to decrypt the
+    // reports. So we distribute the leader encryption keys across all shards.
+    let first_shard_dir = args.output_dir.join(shard_conf_folder(ShardIndex::FIRST));
+    for dest_shard in 1..args.shard_count() {
+        let dest_shard_dir = args.output_dir.join(shard_conf_folder(dest_shard));
+        for helper_id in 1..=3 {
+            fs::copy(
```

Comment:
the previous version of this code just referred to the leader shards conf. We need to have an actual copy of the files?

Reply:
I think the previous version didn't really work because it hasn't created the appropriate files for all shards to be initialized correctly. The structure was something like this

```bash
shard0/
        h1.pem
shard1/
        h2.pem
...
```

instead of 

```bash
   shard0/
        h1.pem,
        h2.pem,
        h3.pem
        ...
```

as each shard needs its own configuration to be complete, they need to have access to encryption keys in their folder